junit-runtime/src/main/scala/org/junit/Assert.scala

```diff
@@ -0,0 +1,349 @@
+/*
+ * Ported from https://github.com/junit-team/junit
+ */
+package org.junit
+
+import org.junit.internal.InexactComparisonCriteria
+import org.junit.internal.ExactComparisonCriteria
+import org.hamcrest.Matcher
+import org.hamcrest.MatcherAssert
+
+object Assert {
+  def assertTrue(message: String, condition: Boolean): Unit =
+    if (!condition) fail(message)
+
+  def assertTrue(condition: Boolean): Unit =
+    assertTrue(null, condition)
+
+  def assertFalse(message: String, condition: Boolean): Unit =
+    assertTrue(message, !condition)
+
+  def assertFalse(condition: Boolean): Unit =
+    assertFalse(null, condition)
+
+  def fail(message: String): Unit = {
+    if (message eq null)
+      throw new AssertionError()
+    throw new AssertionError(message)
+  }
+
+  def fail(): Unit =
+    fail(null)
+
+  def assertEquals[T](message: String, expected: T, actual: T): Unit = {
```

Comment:
I added the polymorphism to allow primitive inputs, which would have been handled be autoboxing in Java. It seems that `Any` is the correct one here.